File: src/com/zypex/piflow/path/WayPoint.java
package com.zypex.piflow.path;

import utils.math.Vector;

public class WayPoint implements PathElement{
    public Double speed = null;
    public Double heading = null;
    public Double rotation = null;
    public Vector position;

    public WayPoint(double x, double y){

    }
    public WayPoint(Vector pos){
        this(pos.x, pos.y);
    }

}

File: src/com/zypex/piflow/path/WayPoint.kt
package com.zypex.piflow.path

import com.zypex.piflow.math.Vector
import piflow.path.PathElement

class WayPoint() : PathElement {
    var speed: Double? = null
    var heading: Double? = null
    var rotation: Double? = null
    var velocity: Vector? = null

    private var _t: Double? = null
    var t: Double?
        get() = _t
        internal set(d){
            _t = d
        }

    val initialized: Boolean
        get() = t == null

    var isStopPoint: Boolean = false


}